pkgs/servers/sql/postgresql/ext/vectorchord/package.nix

```diff
@@ -0,0 +1,83 @@
+{
+  buildPgrxExtension,
+  cargo-pgrx_0_12_9,
+  clang_18,
+  fetchFromGitHub,
+  lib,
+  nix-update-script,
+  nixosTests,
+  postgresql,
+  replaceVars,
+  rustPlatform,
+}:
+let
+  # Upstream recommends clang 18
+  clang = clang_18;
+  rustPlatform' = rustPlatform // {
+    bindgenHook = rustPlatform.bindgenHook.override { inherit clang; };
+  };
+in
+(buildPgrxExtension.override {
+  # Upstream only works with a fixed version of cargo-pgrx for each release,
+  # so we're pinning it here to avoid future incompatibility.
+  cargo-pgrx = cargo-pgrx_0_12_9;
+  rustPlatform = rustPlatform';
+})
+  rec {
+    inherit postgresql;
```

Comment:
Why?

Reply:
You mean why are we passing `postgresql` to `buildPgrxExtension`? The function requires it, since it needs to bind against a specific postgres version: https://github.com/NixOS/nixpkgs/blob/a13cc0bc42a89a025f545c4a17f25e6300710f39/pkgs/development/tools/rust/cargo-pgrx/buildPgrxExtension.nix#L44-L45